tests/inventory/pipelines/base_pipeline_test.py

```diff
@@ -0,0 +1,103 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests the base pipeline."""
+
+
+from google.apputils import basetest
+import mock
+
+# pylint: disable=line-too-long
+from google.cloud.security.common.data_access import dao
+from google.cloud.security.common.data_access import errors as data_access_errors
+from google.cloud.security.common.gcp_api import cloud_resource_manager as crm
+from google.cloud.security.common.gcp_api import errors as api_errors
+from google.cloud.security.common.util import parser
+from google.cloud.security.common.util.log_util import LogUtil
+from google.cloud.security.inventory import errors as inventory_errors
+from google.cloud.security.inventory.pipelines import load_projects_pipeline
+from tests.inventory.pipelines.test_data.fake_projects import EXPECTED_LOADABLE_PROJECTS
+# pylint: enable=line-too-long
+
+
+class BasePipelineTest(basetest.TestCase):
+    """Tests for the base pipeline.
+    
+    Because base pipeline is an abstract class, there is no way to instantiate
+    it for testing directly.  So, will test it by a pipeline that implements it.
+    """
+
+    def setUp(self):
+        """Set up."""
+
+        self.cycle_timestamp = '20001225T120000Z'
+        self.configs = {'organization_id': '66666',
+                        'max_crm_api_calls_per_100_seconds': 400,
+                        'db_name': 'forseti_security',
+                        'db_user': 'sqlproxy',
+                        'db_host': '127.0.0.1',
+                        'email_sender': 'foo.sender@company.com', 
+                        'email_recipient': 'foo.recipient@company.com',
+                        'sendgrid_api_key': 'foo_email_key',}
+        self.mock_crm = mock.create_autospec(crm.CloudResourceManagerClient)
+        self.mock_dao = mock.create_autospec(dao.Dao)
+        self.mock_parser = mock.create_autospec(parser)
+        self.pipeline = (
+            load_projects_pipeline.LoadProjectsPipeline(
+                self.cycle_timestamp,
+                self.configs,
+                self.mock_crm,
+                self.mock_dao))
+
+    def test_data_are_loaded(self):
+        """Test that data are loaded."""
+
+        resource_name = 'foo_resource'
+        self.pipeline._load(resource_name,
+                            EXPECTED_LOADABLE_PROJECTS)
+
+        self.pipeline.dao.load_data.assert_called_once_with(
+            resource_name,
+            self.pipeline.cycle_timestamp,
+            EXPECTED_LOADABLE_PROJECTS)       
+
+    def test_load_errors_are_handled(self):
+        """Test that errors are handled when loading."""
+
+        self.pipeline.dao.load_data.side_effect = (
+            data_access_errors.MySQLError('error error', mock.MagicMock()))
+
+        self.assertRaises(inventory_errors.LoadDataPipelineError,
```

Comment:
Nit: The context manager version of assertRaises is easier to read for tests like this one.

with self.assertRaises(...):
  self.pipeline._load(...)